Sources/Basic/FileSystem.swift

```diff
@@ -101,6 +101,9 @@ public protocol FileSystem {
     
     /// Check whether the given path is accessible and a file.
     func isFile(_ path: AbsolutePath) -> Bool
+    
+    /// Check to determine whether the given path is an accessible and executable file
+    func isExecutableFile(_ path: AbsolutePath) -> Bool
```

Comment:
Please create a separate PR for this change. We try to keep all orthogonal changes (especially API addition in Basic) in different PRs.